sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -164,6 +164,7 @@ class StreamSuite extends StreamTest {
       /* -- batch 0 ----------------------- */
       // Add some data in batch 0
       AddData(inputData, 1, 2, 3),
+      EnsureManualClockInWaitingState,
```

Comment:
This is necessary. Otherwise, the manual clock may be advanced too early and then `ManualClock.waitTillTime` will block forever.